src/psalm.php

```diff
@@ -189,7 +192,18 @@ function ($arg) use ($valid_long_options, $valid_short_options) {
 if (!array_key_exists('use-ini-defaults', $options)) {
     ini_set('display_errors', 'stderr');
     ini_set('display_startup_errors', '1');
-    ini_set('memory_limit', (string) (8 * 1024 * 1024 * 1024));
+
+    $memoryLimit = (8 * 1024 * 1024 * 1024);
+
+    if (array_key_exists('memory-limit', $options)) {
+        $memoryLimit = $options['memory-limit'];
+
+        if (!is_scalar($memoryLimit)) {
+            throw new ConfigException('Invalid memory limit specified.');
```

Comment:
I'm not sure if this case is even possible -- however without the scalar check, psalm complained about the following: 

```
ERROR: PossiblyInvalidCast - src/psalm.php:202:38 - list<mixed> cannot be cast to string (see https://psalm.dev/190)
    ini_set('memory_limit', (string) $memoryLimit);
```

Reply:
`getopt()` will return a list if you specify the option multiple times. 
E.g `psalm --memory-limit=1 --memory-limit=2` will result in
```
array(1) {
  'memory-limit' =>
  array(2) {
    [0] =>
    string(1) "1"
    [1] =>
    string(1) "2"
  }
}
```

Reply:
today I learned!